File: AdventOfCode.FSharp/Examples/Day0.fs
namespace AdventOfCode.FSharp.Examples

open AdventOfCode.Core
open AdventOfCode.FSharp.Shared
open Xunit

module Day0 =
    let Solve =
        Seq.fold (fun acc item -> acc * int64 item) 1L

    [<Fact>]
    let ``Solves Example`` () =
        let exampleInput = @"foo"

        let actual = Solve exampleInput
        Assert.Equal(1256742L, actual)

type Day0(input: string) =
    interface ISolver with
        member this.SolvePartOne() = Day0.Solve input
        member this.SolvePartTwo() = failwith "NotImplemented"

File: AdventOfCode.CSharp/Examples/Day0.cs
using AdventOfCode.Core;
using Xunit;

namespace AdventOfCode.CSharp.Examples;

public class Day0 : Solver {
    public Day0(string? input = null) : base(input) { }

    public override long SolvePartOne() {
        return Input.ToCharArray().Aggregate(1L, (current, c) => current * c);
    }

    [Fact]
    public void SolvesExample() {
        const string? exampleInput = @"foo";

        var solver = new Day0(exampleInput);
        var actual = solver.SolvePartOne();

        Assert.Equal(1256742, actual);
    }
}